engine/graphics/src/opengl/graphics_opengl.cpp

```diff
@@ -4412,37 +4445,51 @@ static void LogFrameBufferError(GLenum status)
 
     static uint16_t OpenGLGetTextureWidth(HTexture texture)
     {
-        return GetAssetFromContainer<OpenGLTexture>(g_Context->m_AssetHandleContainer, texture)->m_Width;
+        ScopedLock lock(g_Context->m_AssetHandleContainerMutex);
```

Comment:
I added the lock to all of these functions as well as a check if the returned texture is zero or not, it might be the case that we try to use a texture that has been deleted but we still have a handle around. Before these guards we would get an engine crash. I _don't_ think we have had such crashes before, but at least to me it makes sense to fully guard the handle container, especially if we want to move more of these functions to the DMSDK